src/Storage/Field/Type/DateType.php

```diff
@@ -38,7 +51,24 @@ public function query(QueryInterface $query, ClassMetadata $metadata)
     }
 
     /**
-     * @inheritdoc
+     * {@inheritdoc}
+     */
+    public function persist(QuerySet $queries, $entity, EntityManager $em = null)
+    {
+        $key = $this->mapping['fieldname'];
+        $value = $entity->get($key);
+
+        if (!$value instanceof \DateTime && $value !== null) {
+            $value = new Carbon($value);
+            $value::setToStringFormat('Y-m-d');
```

Comment:
Probably shouldn't set the static string format here, because it will apply globally. Can you call `toString` manually with that format instead?